Lib/test/test_tstring.py

```diff
@@ -0,0 +1,441 @@
+import ast
+import unittest
+
+from string.templatelib import Template, Interpolation
+
+
+def convert(value, conversion):
+    if conversion == "a":
+        return ascii(value)
+    elif conversion == "r":
+        return repr(value)
+    elif conversion == "s":
+        return str(value)
+    return value
+
+
+def f(template):
+    parts = []
+    for item in template:
+        match item:
+            case str() as s:
+                parts.append(s)
+            case Interpolation(value, _, conversion, format_spec):
+                value = convert(value, conversion)
+                value = format(value, format_spec)
+                parts.append(value)
+    return "".join(parts)
+
+
+class TestTString(unittest.TestCase):
+    def assertAllRaise(self, exception_type, regex, error_strings):
+        for s in error_strings:
+            with self.subTest(s=s):
+                with self.assertRaisesRegex(exception_type, regex):
+                    eval(s)
+
+    def test_template_basic_creation(self):
+        # Simple t-string creation
+        t = t"Hello, world"
+        self.assertTrue(isinstance(t, Template))
+        self.assertEqual(t.strings, ("Hello, world",))
+        self.assertEqual(len(t.interpolations), 0)
+        self.assertEqual(f(t), "Hello, world")
+
+        # Empty t-string
+        t = t""
+        self.assertEqual(t.strings, ("",))
+        self.assertEqual(len(t.interpolations), 0)
+        self.assertEqual(f(t), "")
+
+        # Multi-line t-string
+        t = t"""Hello,
+world"""
+        self.assertEqual(t.strings, ("Hello,\nworld",))
+        self.assertEqual(len(t.interpolations), 0)
+        self.assertEqual(f(t), "Hello,\nworld")
+
+    def test_template_creation_interleaving(self):
```

Comment:
This test might fit better in a test file for `string.templatelib`. This file can then test only the syntax (t-strings) and the templatelib tests can test the Template and Interpolation classes.

Reply:
Also there should be more tests for the Template and Interpolation classes, e.g. make sure they can't be subclassed or mutated.

Reply:
If splitting, we could move the syntax tests to `test_syntax`, which would mean `test_tstring` could be removed entirely / renamed to `test_string/test_templatelib`?

Reply:
I think it's fine to have a test file dedicated to testing this piece of syntax, similar to how we have `test_type_annotations.py` and `test_annotationlib.py`.

Reply:
@AA-Turner took care of this.